sdk/tests/conformance/reading/read-pixels-test.html

```diff
@@ -206,37 +199,113 @@
         type: gl.UNSIGNED_SHORT_4_4_4_4,
         dest: new Uint16Array(1),
       },
+    ];
+
+    const FORMATS = [
       {
+        format: gl.RGBA,
+        channels: 4,
+      }, {
+        format: gl.RGB,
+        channels: 3,
+      }, {
+        format: gl.LUMINANCE_ALPHA,
+        channels: 2,
+      }, {
         format: gl.ALPHA,
-        type: gl.UNSIGNED_BYTE,
-        dest: new Uint8Array(1),
-      }
+        channels: 1,
+      }, {
+        format: gl.LUMINANCE,
+        channels: 3,
+      },
     ];
-    if (contextVersion > 1) {
-      combinations = combinations.concat([
+    if (contextVersion >= 2) {
+      FORMATS.push(
         {
           format: gl.RED,
-          type: gl.UNSIGNED_BYTE,
-          dest: new Uint8Array(1),
-        },
-        {
+          channels: 1,
+        }, {
+          format: gl.RG,
+          channels: 1,
+        }, {
           format: gl.RGBA_INTEGER,
-          type: gl.UNSIGNED_INT,
-          dest: new Uint32Array(4),
-        },
+          channels: 4,
+        }, {
+          format: gl.RGB_INTEGER,
+          channels: 3,
+        }, {
+          format: gl.RG_INTEGER,
+          channels: 2,
+        }, {
+          format: gl.RED_INTEGER,
+          channels: 1,
+        }
+      );
+    }
+
+    // -
+
+    const TYPES = [
+      {
+        type: gl.UNSIGNED_BYTE,
+        ctor: Uint8Array,
+      }, {
+        type: gl.BYTE,
+        ctor: Int8Array,
+      }, {
+        type: gl.UNSIGNED_SHORT,
+        ctor: Uint16Array,
+      }, {
+        type: gl.SHORT,
+        ctor: Int16Array,
+      }, {
+        type: gl.UNSIGNED_INT,
+        ctor: Uint32Array,
+      }, {
+        type: gl.INT,
+        ctor: Int32Array,
+      }, {
+        type: gl.FLOAT,
+        ctor: Float32Array,
+      }
+    ];
+
+    if (contextVersion >= 2) {
+      TYPES.push(
         {
-          format: gl.RGBA_INTEGER,
-          type: gl.INT,
-          dest: new Int32Array(4),
+          type: gl.HALF_FLOAT,
+          ctor: Uint16Array,
         }
-      ]);
+      );
+    }
+
+    const ext = gl.getExtension('OES_texture_half_float');
+    if (ext) {
+      TYPES.push(
+        {
+          type: gl.HALF_FLOAT_OES,
```

Comment:
This should be `ext.HALF_FLOAT_OES`. Please test the WebGL 1.0 version of this modification. e.g.:

http://localhost:8080/sdk/tests/conformance/reading/read-pixels-test.html